snuba/clusters/storage_sets.py

```diff
@@ -0,0 +1,11 @@
+from typing import Mapping, Set
+
+
+STORAGE_SETS: Mapping[str, Set[str]] = {
+    "events": {"errors", "events", "groupedmessages"},
+    "groupassignees": {"groupassignees"},
+    "outcomes": {"outcomes_raw", "outcomes_hourly"},
+    "querylog": {"querylog"},
+    "sessions": {"sessions_raw", "sessions_hourly"},
+    "transactions": {"transactions"},
+}
```

Comment:
I think we can improve types here.
If you create a type alias for StorageSet (that maps to string) and a type alias for the storage_key (that maps to string), then we can properly qualify storage set names and storage keys in methods signatures instead of using strings and the type checker can enforce that.

But beyond that, referring to storages by storage_key here is calling for integrity issues (a typo here would show up only in production). Why don't you define storage keys either through constants or through an Enum ? So the type checker can enforce we only reference valid ones. Also you could reference the instance of the storage, but that would bake in the assumption that storages are singleton at runtime which we do not need.